src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -1407,6 +1408,31 @@ ActiveShardPlacementList(uint64 shardId)
 }
 
 
+/*
+ * ShardPlacementListWithoutOrphanedPlacements returns shard placements exluding
+ * the ones that are orphaned, because they are marked to be deleted at a later
+ * point (shardstate = 4).
+ */
+List *
+ShardPlacementListWithoutOrphanedPlacements(uint64 shardId)
```

Comment:
Not thrilled about the name, but we can revisit once we remove the inactive shard placement logic.